### PR TITLE
Bump js libs and use @nextcloud/vue 7.0.0

### PR DESCRIPTION
**Pull request by julien-nc**
* Bump most js libs (except jest-related stuff)
* Adjust to new webpack config (use full file names in imports)
* Use `@nextcloud/vue` 7.0.0 in which all component names are prefixed with `Nc`
* Adjust to `NcEmptyContent` new stuff

The main reason for this is to get the new NcAvatar component which should adapt to the dark theme as you are using it in a classic fashion. (I mean you are not using some crazy css customization :grin:).

I don't have a local env with OpenProject set up so I could not try everything.
Related workpackage: https://community.openproject.org/projects/nextcloud-integration/work_packages/47126/activity

### CODE REVIEW

**Comment by julien-nc** [comment]
Summary of the jest tests failure reasons:
* NcPopover has changed
* TextInput.vue failure is just about the `data-v-XXX` attribute that has changed
* NcAvatar has changed
* ProjectsTab.vue to be checked in real conditions

If everything works fine, let's update the snapshots.

**Comment by kiranparajuli589** [comment]
I'm checking|testing with this PR.

**Comment by kiranparajuli589** [comment]
With:
- latest master of `NextCloud` and this branch for the `integrationn_app`
- `dark` theme enabled

**Dashboard widget looks like this:**
Screenshot from 2022-10-14 12-03-43

Found this https://github.com/nextcloud/nextcloud-vue-dashboard/blob/master/src/DashboardWidgetItem.vue component using `Avatar` component. Is this the new|updated one?

**Comment by julien-nc** [comment]
Oh right, the dashboard components are now included in `@nextcloud/vue`. I changed that and injected dummy content in the widget:
image

**Comment by kiranparajuli589** [comment]
Now, it works as expected (tested with latest master of NC). I've also pushed the updated snapshots. There was a small test adjustment needed for the SearchInput spec now working properly (loading icon selector got changed).

**Comment by individual-it** [comment]
I cannot make the dark avatars work, tried to reduce cache in https://github.com/nextcloud/integration_openproject/blob/333674a808f5269acfb891c5992f72c8b636ef4a/lib/Controller/OpenProjectAPIController.php#L117
restarted webserver, no change.

image

what do I miss?

**Comment by kiranparajuli589** [comment]
Difference: at the profile avatar, the link is like:
Screenshot from 2022-11-03 12-00-29

And at out dashboard the avatar has the link like:
Screenshot from 2022-11-03 12-00-46

Should we manage that `dark` word in the link programatically when managing items for the dashboard, or there can be any more efficient NC way. @julien-nc @individual-it

**Comment by individual-it** [comment]
@kiranparajuli589 I don't know of any other way, so lets add it programmatically

File: tests/jest/views/Dashboard.spec.js [hunk 1]
```diff
@@ -64,7 +64,8 @@ describe('Dashboard.vue', () => {
 		expect(wrapper.vm.items).toMatchSnapshot()
 		axiosSpy.mockRestore()
 	})
-	describe('mark as read', () => {
+
+	describe.skip('mark as read', () => {
```
Comment (SwikritiT) [inline tests/jest/views/Dashboard.spec.js:68]: I'm skipping this test that fails in CI for now, so as not to block this PR. I'll fix it separately

**Review by individual-it — APPROVED** [review]
all looks good to me
:+1:

**Pull request merged**